tests/unit/test_base.py

```
@@ -341,3 +359,195 @@ def test_request_non_retryable_error(mock_request, mock_endpoint, base_client):
+                    client.request(mock_endpoint)
+
+        # Callback should have been called for each attempt
+        assert len(callback_calls) >= 1
+        # At least the last call should have success=False
+        last_call = callback_calls[-1]
+        assert last_call["success"] is False
+
+    def test_metrics_callback_exception_doesnt_break_request(self, mock_endpoint):
+        """Test that a failing metrics callback doesn't break the request."""
+
+        def failing_callback(**kwargs):
```

Ruff (0.14.13)

Unused function argument: `kwargs`

(ARG001)